doc/source/cluster/kubernetes/user-guides/kuberay-gcs-ft.md

```diff
@@ -0,0 +1,261 @@
+# 2023-09-07 17:31:17,875 INFO worker.py:1612 -- Connected to Ray cluster. View the dashboard at http://10.244.0.21:8265
+# 2
+```
+
+```{admonition} The detached actor is always on the worker Pod in this example.
+The head Pod's `rayStartParams` is set to `num-cpus: "0"`.
+Hence, no tasks or actors will be scheduled on the head Pod.
+```
+
+With GCS fault tolerance enabled, we can still access the detached actor after the GCS process is dead and restarted.
+Note that the fault tolerance doesn't persist the actor's state.
+The reason why the result is 2 instead of 1 is that the detached actor is on the worker Pod which is always running.
+On the other hand, if the head Pod hosts the detached actor, the `increment_counter.py` script yields a result of 1 in this step.
+
+### Step 9: Delete the Kubernetes cluster
+
+```sh
+kind delete cluster
+```
+
+## KubeRay GCS fault tolerance configurations
+
+The [ray-cluster.external-redis.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.external-redis.yaml) used in the quickstart example contains detailed comments about the configuration options.
+***It's recommended to read this section in conjunction with the YAML file.***
+
+### 1. Enable GCS fault tolerance
+
+* **`ray.io/ft-enabled`**: Add `ray.io/ft-enabled: "true"` annotation to the RayCluster custom resource to enable GCS fault tolerance.
+    ```yaml
+    kind: RayCluster
+    metadata:
+    annotations:
+        ray.io/ft-enabled: "true" # <- Add this annotation to enable GCS fault tolerance
+    ```
+
+### 2. Connect to an external Redis
+
+* **`redis-password`** in head's `rayStartParams`:
+Use this option to specify the password for the Redis service, thus allowing the Ray head to connect to it.
+In the [ray-cluster.external-redis.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.external-redis.yaml), it uses an environment variable `REDIS_PASSWORD` to store the password from a Kubernetes secret.
+    ```yaml
+    rayStartParams:
+      redis-password: $REDIS_PASSWORD
+    template:
+      spec:
+        containers:
+          - name: ray-head
+            env:
+              # This environment variable is used in the `rayStartParams` above.
+              - name: REDIS_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: redis-password-secret
+                    key: password
+    ```
+
+* **`RAY_REDIS_ADDRESS`** environment variable in head's Pod:
+Ray reads the `RAY_REDIS_ADDRESS` environment variable to establish a connection with the Redis server.
+In the [ray-cluster.external-redis.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.external-redis.yaml), it uses the `redis` Kubernetes ClusterIP service name as the connection point to the Redis server. The ClusterIP service is also created by the YAML file.
```

Comment:
Similar comment to above.

Reply:
Updated https://github.com/ray-project/ray/pull/39450/commits/c2ec123ef358aebbb835a15afc40744d5e05f471